src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,231 @@
+
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Methods to bind in 'this' context
+ */
+const METHODS_TO_BIND = [
+  'getDetailLink',
+  'getDetailMessages',
+  'getFixedPart',
+  'getToggledPart',
+  'toggleExpanded'
+];
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ *
+ * @example <caption>How to use the CollapsibleErrorMessage</caption>
+ *   <CollapsibleErrorMessage
+ *
+ *        className="error-for-modal"
+ *
+ *        details={[ 'an', 'array', 'of', 'detailed errors' ]}
+ *        message="The message to display"
+ *
+ *        onToggle={function(isToggled) {
+ *          // Do something
+ *        }}
+ *
+ *      />
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+    // Bind methods in context
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+  }
+
+  /**
+   * Set the default expandedg from the properties
+   */
+  componentWillMount() {
+    this.setState({
+      expanded: this.props.expanded
+    });
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    this.setState({
+      expanded: !this.state.expanded
+    });
+
+    // Call onToggle callback
+    this.props.onToggle(this.state.expanded);
```

Comment:
Damn, you are right! To my defence I initially didn't plan to add an argument to the callback, but I thought that having it can only be beneficial 😄 
Fixed!